src/test/java/com/commercetools/sync/commons/utils/completablefutureutils/MapValuesToFutureOfCompletedValuesStreamTest.java

```diff
@@ -0,0 +1,219 @@
+        final Stream<String> result = future.join();
+        assertThat(result).containsExactly("fooPOSTFIX");
+        assertThat(result).isInstanceOf(Stream.class);
+    }
+
+    @Test
+    public void single_ListWithDiffTypeMapping_ReturnsFutureOfStreamOfMappedValue() {
+        final CompletableFuture<Stream<Integer>> future = mapValuesToFutureOfCompletedValues(
+            singletonList("foo"), element -> completedFuture(element.length()));
+        final Stream<Integer> result = future.join();
+        assertThat(result).containsExactly(3);
+        assertThat(result).isInstanceOf(Stream.class);
+    }
+
+    @Test
+    public void single_SetWithDiffTypeMapping_ReturnsFutureOfStreamOfMappedValue() {
+        final CompletableFuture<Stream<Integer>> future = mapValuesToFutureOfCompletedValues(
+            singleton("foo"), element -> completedFuture(element.length()));
+        final Stream<Integer> result = future.join();
+        assertThat(result).containsExactly(3);
+        assertThat(result).isInstanceOf(Stream.class);
+    }
+
+
+    /**
+     * * List to Stream : Same Type Mapping : Multiple Values : no duplicates.
+     * * Set to Stream : Same Type Mapping : Multiple Values : no duplicates.
+     * * List to Stream : Diff Type Mapping : Multiple Values : no duplicates.
+     * * Set to Stream : Diff Type Mapping : Multiple Values : no duplicates.
+     */
+
+    @Test
+    public void multiple_ListWithSameTypeMappingNoDuplicates_ReturnsFutureOfStreamOfMappedValues() {
+        final CompletableFuture<Stream<String>> future = mapValuesToFutureOfCompletedValues(
+            asList("foo", "bar"), element -> completedFuture(element.concat("POSTFIX")));
+        final Stream<String> result = future.join();
+        assertThat(result).containsExactly("fooPOSTFIX", "barPOSTFIX");
+        assertThat(result).isInstanceOf(Stream.class);
+    }
+
+    @Test
+    public void multiple_SetWithSameTypeMappingNoDuplicates_ReturnsFutureOfStreamOfMappedValues() {
+        final Set<String> set = new HashSet<>();
+        set.add("foo");
+        set.add("bar");
+
+        final CompletableFuture<Stream<String>> future = mapValuesToFutureOfCompletedValues(set,
+            element -> completedFuture(element.concat("POSTFIX")));
+        final Stream<String> result = future.join();
+        assertThat(result).containsExactlyInAnyOrder("fooPOSTFIX", "barPOSTFIX");
+        assertThat(result).isInstanceOf(Stream.class);
+    }
+
+    @Test
+    public void multiple_ListWithDiffTypeMappingNoDuplicates_ReturnsFutureOfStreamOfMappedValues() {
+        final CompletableFuture<Stream<Integer>> future = mapValuesToFutureOfCompletedValues(
+            asList("john", "smith"), element -> completedFuture(element.length()));
+        final Stream<Integer> result = future.join();
+        assertThat(result).containsExactly(4, 5);
+        assertThat(result).isInstanceOf(Stream.class);
```

Comment:
amazing test :-)

Reply:
???
why do you assert instance of `Stream` variable is instance of `Stream`?

Reply:
oh, copy-paste mistake. thanks: 630741b